### PR TITLE
Changes to comply MS documentation

### PR DESCRIPTION
**Pull request by risototh**
Hi, 

we had many problem reading Outlook messages with hfig/MAPI library and debugged it down to this library. I made an extensive code review and fixed (at least hope so) issues by comparing it to the MS-CFB documentation. 

I'm not sure, what was the intention with the `readSignedInt4()` function, but in the documentation, there is no place, where the data can be handled as _signed_. This of course created many _invalid_ pointers in the FAT tables on larger files - where the pointers turned around to the negative number part. 
Also it is better to stick to the special meanings values (empty sector, end of chain, etc.) as they are described in the documentation - e.g. to use 0xFFFFFFFE instead of -2...
So, I defined bunch of constants and adjusted the whole code to use them. Now it seems to work like a charm.

I also noticed, that during the file write, there was an invalid _minor version_ value. The documentation says, that it SHOULD be 0x003E, thus I don't know the source of the 0x3b, that was in code. With this I also reformatted the writing code to be readable a bit better, value by value, as described in the documentation. 

Last thing I changed is the extraction of the executive part of the `read()` function to the `readStream()` as it can be used by another higher level library, that uses the streams and not files. One example is the hfig/MAPI library, where the whole `read()` function was replicated, and I think, that this is a bad option as maintaining the sync between the two libraries can be a nightmare. Basically, the `read()` will just use the `readStream()` and decorate it with the opening of the desired file. Not a huge change, but very useful. 

After merging this to master and creating a new RC, I will also push a pull request to the hfig/MAPI.

Cheers.

### CODE REVIEW

**Comment by risototh** [comment]
after update of the test data (changed the minor version in the Example.bin), the checks have passed :)

File: OLE.php [hunk 1]
```diff
@@ -133,68 +143,86 @@ function read($file)
...
+    {
+        $this->_file_handle = $stream;
+
+        $signature = fread($stream, 8);
+        if (OLE_CFB_SIGNATURE != $signature) {
             return $this->raiseError("File doesn't seem to be an OLE container.");
         }
-        fseek($fh, 28);
-        if (fread($fh, 2) != "\xFE\xFF") {
+        fseek($stream, 28);
```
Comment (sanmai) [inline OLE.php:165]: I wish this didn't change `$fh` to `$stream` to keep the git blame aligned. This is nitpicking, truly.

**Review by sanmai — APPROVED** [review]
Great work! 

There's a minor comment. If you can fix it, I'll be happy to merge this ASAP.

File: OLE.php (hunk 1 above)
Comment (risototh) [reply OLE.php:165]: Not sure, if I understand it. Should I change the $stream in the readStream() function to $fh?

Comment (sanmai) [reply OLE.php:165]: If you can do that, that'd be great. This way `git blame` won't show you as the author of lines where `$fh` was changed to `$stream`, instead pointing at earlier commits.

Comment (risototh) [reply OLE.php]: Changed as you requested. Thank you.

Comment (sanmai) [reply OLE.php]: Excellent! And I just made a release with this fix.

**Comment by risototh** [comment]
Thank you!

**Pull request merged**